lib/ansible/modules/windows/win_disk_facts.ps1

```diff
@@ -0,0 +1,181 @@
+                        type = $part.Type
+                        drive_letter = "$([string]$part.DriveLetter)"
+                        transition_state = $part.TransitionState
+                        offset = $part.Offset             
+                        hidden = $part.IsHidden
+                        shadow_copy = $part.IsShadowCopy
+                        guid = "$([string]$part.Guid)"
+                        access_paths = "$([string]$part.AccessPaths)"
+                    }
+                    if ($disks.PartitionStyle -eq "GPT") {
+                        $partition_results.gpt_type = $part.GptType
+                        $partition_results.no_default_driveletter = $part.NoDefaultDriveLetter
+                    } elseif ($disks.PartitionStyle -eq "MBR") {
+                        $partition_results.mbr_type = $part.MbrType
+                        $partition_results.active = $part.IsActive
+                    }
+                    $vols = Get-Volume -Partition $part -ErrorAction SilentlyContinue
+                    if ($vols) {
+                        $partition_results["volumes"]  += @()
+                        foreach ($vol in $vols) {
+                            $volume_results  = @{
+                                size = "$($vSize = "{0:N3}" -f ($vol.Size / 1GB))$($vSize)gb"
+                                size_remaining = "$($vSizeRe = "{0:N3}" -f ($vol.SizeRemaining / 1GB))$($vSizeRe)gb"
+                                type = $vol.FileSystem
+                                label = $vol.FileSystemLabel
+                                health_status = $vol.HealthStatus
+                                drive_type = $vol.DriveType
+                                object_id = $vol.ObjectId
+                                path = $vol.Path
+                            }
+                            if ([System.Environment]::OSVersion.Version.Major -ge 10) {
+                                $volume_results.allocation_unit_size = "$($vol.AllocationUnitSize /1KB)kb"
+                            } else {
+                                $volPath = ($vol.Path.TrimStart("\\?\")).TrimEnd("\")
+                                $BlockSize = (Get-CimInstance -Query "SELECT BlockSize FROM Win32_Volume WHERE DeviceID like '%$volPath%'" -ErrorAction SilentlyContinue | Select-Object BlockSize).BlockSize
+                                $volume_results.allocation_unit_size = "$($BlockSize / 1KB)kb"
+                            }
+                            $partition_results.volumes  += $volume_results
+                        }
+                    }
+                $disk_results.partitions += $partition_results
+                }
+            }
+            $result.ansible_facts.ansible_disk += $disk_results
+        }
+    } else {
+            $result.ansible_facts.ansible_disk.total_disks_found = "0"
+            Fail-Json -obj $result -message "No disks could be found on the target"
```

Comment:
Will this ever happen, is it when you run as a non-administrator?

Reply:
Removed it